library/src/G4LatticeReader.cc

```diff
@@ -392,12 +394,27 @@ G4bool G4LatticeReader::ProcessEulerAngles(const G4String& name) {
     G4cerr << "G4LatticeReader: Unknown rotation matrix " << name << G4endl;
     return false;
   }
-
   G4double degOrRad = ProcessUnits("Angle");
   pLattice->AddValley(phi*degOrRad, theta*degOrRad, psi*degOrRad);
   return psLatfile->good();
 }
 
+
+// Read miller indices (milleri, millerj, millerk) for named rotation matrix
+
+G4bool G4LatticeReader::ProcessValleyDirection() {
+  G4double milleri=0., millerj=0., millerk=0.;
+  *psLatfile >> milleri >> millerj >> millerk;
+  if (verboseLevel>1)
+    G4cout << " ProcessValleyDirection "  << milleri << " " 
+	   << millerj << " " << millerk << G4endl;
+
+  G4ThreeVector valleyDirVec(milleri,millerj,millerk);
+  pLattice->AddValley(valleyDirVec);
```

Comment:
This is where you should call `pLattice->AddValley(-valleyDirVec);` to get the anti-valley.